packages/platform-core/src/usage-data/account_id_fetcher.ts

```diff
@@ -1,22 +1,32 @@
 import { GetCallerIdentityCommand, STSClient } from '@aws-sdk/client-sts';
 
+const NO_ACCOUNT_ID = '000000000000';
 /**
  * Retrieves the account ID of the user
  */
 export class AccountIdFetcher {
+  private accountId: string | undefined;
   /**
    * constructor for AccountIdFetcher
    */
   constructor(private readonly stsClient = new STSClient()) {}
   fetch = async () => {
-    const stsResponse = await this.stsClient.send(
-      new GetCallerIdentityCommand({})
-    );
-    if (stsResponse && stsResponse.Account) {
-      return stsResponse.Account;
+    if (this.accountId) {
+      return this.accountId;
+    }
+    try {
+      const stsResponse = await this.stsClient.send(
+        new GetCallerIdentityCommand({})
+      );
+      if (stsResponse && stsResponse.Account) {
+        this.accountId = stsResponse.Account;
+        return this.accountId;
+      }
+      // We failed to get the account Id. Most likely the user doesn't have credentials
+      return NO_ACCOUNT_ID;
```

Comment:
Not sure how I feel about a string value that means "not found". Could we return undefined instead of throwing and update callers to handle a potentially undefined result?

Reply:
This is specifically scoped in the `usage-data`. Challenge is the telemetry pipeline not supporting the undefined values.

Reply:
Should the value be the string literal 'NO_ACCOUNT_ID' then? Instead of something that might accidentally be parsed as a number

Reply:
Let me test if the pipeline is ok with a non-decimal value.

Reply:
Yup, it works e2e. Updated the PR.